kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -370,6 +383,61 @@ extension [A, S, E](effect: A < (Abort[E] & S))
             case Result.Success(v) => v
         }
 
+    /** Recovers from an Abort failure by applying the provided function.
+      *
+      * This method allows you to handle failures in an Abort effect and potentially continue the computation with a new value. It only
+      * handles failures of type E and leaves panics unhandled (Abort[Nothing]).
+      *
+      * @param onSuccess
+      *   A function that takes the success value of type A and returns a new computation
+      * @param onFail
+      *   A function that takes the failure value of type E and returns a new computation
+      * @param v
+      *   The original computation that may fail
+      * @return
+      *   A computation that either succeeds with the original value or the recovered value
+      */
+    def foldAbort[B, S1](
+        onSuccess: A => B < S1,
+        onFail: E => B < S1
+    )(
+        using
+        ct: SafeClassTag[E],
+        fl1: Flat[A],
+        fl2: Flat[B],
+        fr: Frame
+    ): B < (S & S1) =
+        Abort.foldOrThrow(onSuccess, onFail)(effect)
```

Comment:
Note that in kyo-combinators, I'm throwing `Panic` exceptions by default for all error-handling methods that ignore panic *when handling for the full `Abort` type*. When using `forAbort[E]`, these methods will leave `Panic` unhandled and eventually reduce to `Abort[Nothing]`.

Reply:
Is it a usability decision? It'd be nice to keep a similar behavior to the Abort APIs

Reply:
Yeah usability.  The point of this PR is to have a way to deal with `Abort` where you don't have to think about panic. Leaving an `Abort[Nothing]` in the pending intersection when you handle success and failure means you have to handle `Abort` a second time which is not great.

On the other hand, if you are only handling a single subtype within the `Abort`, you probably don't want to throw `Panic` exceptions. For this reason, I broke the `Abort` methods into two forms -- one that leaves `Abort[Nothing]` and one that throws (`Abort.fold` vs `Abort.foldOrThrow`, and `Abort.runPartial` and `Abort.runPartialOrThrow`). The `...OrThrow` methods operate on only the whole `Abort` type rather than allowing you to specify a type. This is essentially the same behavior as kyo-combinators.

Reply:
Sounds reasonable. Something to keep in mind is that `Abort[Nothing]` is included in `IO` and `Async` but I think the compiler doesn't automatically de-duplicate the pending effect.

Reply:
Hmm I didn't realize that. In that case, it probably makes more sense to have all of these methods *not* throw and just handle to `Abort[Nothing]`, and to add separate `...OrThrow` methods. Most cases where a user will want to handle `Abort` will be in conjunction with `Async` or `IO` anyways, and this could result in repeatedly throwing and catching the same exceptions, which seems bad.

I'll make the combinators work the same way as `Abort`: `fold`/`foldOrCatch` and `partialResult`/`partialResultOrThrow`.

I'll make `orPanic` reduce `Abort[X]` to `Abort[Nothing]`, and I'll add an `orThrow` method to eliminate `Abort` entirely by throwing exceptions.

Reply:
Done. I feel happy with the current API.